test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4414,6 +4414,23 @@ def _get_pid(self, inst):
             self.pid = None
         return self.pid
 
+    def _update_pid(self, inst):
+        """
+        update pid of given inst
+
+        :param inst: inst to update pid
+        :type inst: object
+        """
+        i = 0
+        while i < 30:
```

Comment:
A few of reasons.  I find that traditionally for loops are meant for iterating.  Python even goes further than C by allowing you to iterate over lists and dictionaries.  I think it's more concise.  The reader of the code can see up front what is going on instead of searching through the loop looking for the iteration statement.  Lastly, it is safer.  If a continue is placed in the wrong place in a while loop, you can go into an infinite loop.  This isn't possible in a for loop since you'll eventually reach the end of your iteration.